module-6/ocp-example.js

```diff
@@ -0,0 +1,157 @@
+    </div>
+  )
+}
+
+// !!! Requirement change: if the country is US we need to insert commas !!!
+// Before: 10000.000
+// After: 10,000.000
+
+// Solution:
+function NumberFormatter({ format, number }) {
+  const adjustedFormat = checkIfUS() ? `0,${format}` : format // 0,0.000
+
+  return (
+    <div>
+      {numeralJs(adjustedFormat)(number)}
+    </div>
+  )
+}
+
+// !!! Requirement change: if the `display negatives in parenthesis` in the settings is set !!!
+// !!! then negatives should be displayed in parenthesis !!!
+// Before: -10000.000
+// After: (10000.000)
+// Before: -10,000.000
+// After: (10,000.000)
+
+// Solution (probably breaking OCP, but who knows):
+const mapStateToProps = state => ({
+  displayNegativesInParenthesis: state.settings.displayNegativesInParenthesis,
+})
+
+connect(mapStateToProps)(
+  function NumberFormatter({ format, number, displayNegativesInParenthesis }) {
+    const locationAdjustedFormat = checkIfUS() ? `0,${format}` : format // 0,0.000
+
+    const settingsAdjustedFormat = displayNegativesInParenthesis
+      ? `(${locationAdjustedFormat})` // (0,0.000)
+      : locationAdjustedFormat
+
+    return (
+      <div>
+        {numeralJs(settingsAdjustedFormat)(number)}
+      </div>
+    )
+  }
+)
+
+// !!! Requirement change: if we are on the `/shop/dinosaur-toys` route we should paint the numbers green !!!
+
+// Solution (probably breaking OCP, but who knows):
+const mapStateToProps = state => ({
+  displayNegativesInParenthesis: state.settings.displayNegativesInParenthesis,
+})
+
+connect(mapStateToProps)(
+  withRouter(
+    function NumberFormatter({
+      format,
+      number,
+      displayNegativesInParenthesis,
+      location, // <- this is new
+    }) {
+      const locationAdjustedFormat = checkIfUS() ? `0,${format}` : format // 0,0.000
+
+      const settingsAdjustedFormat = displayNegativesInParenthesis
+        ? `(${locationAdjustedFormat})` // (0,0.000)
+        : locationAdjustedFormat
+
+      const className = classnames({
+        red: location.pathname.includes('dinosaur-toys')
+      })
+
+      return (
+        <div className={className}>
+          {numeralJs(settingsAdjustedFormat)(number)}
+        </div>
+      )
+    }
+  )
+)
+
+// Solution (probably better in terms of OCP):
+const mapStateToProps = state => ({
+  displayNegativesInParenthesis: state.settings.displayNegativesInParenthesis,
+})
+
+function withAdjustedFormatting(ComponentToWrap) {
+  function LocalizedFormatter({ displayNegativesInParenthesis, ...restProps }) {
+    const locationAdjustedFormat = format => checkIfUS() ? `0,${format}` : format // 0,0.000
+    const settingsAdjustedFormat = format => displayNegativesInParenthesis
+      ? `(${format})` // (0,0.000)
+      : format
+
+    const adjustFormatting = format => settingsAdjustedFormat(locationAdjustedFormat(format))
+
+    return <ComponentToWrap adjustFormatting={adjustFormatting} {...restProps} />
```

Comment:
ja bym zrobił `{...props} adjustFormatting={adjustFormatting}` żeby przypadkiem nikt nie napisał tego (ale to paranoja)

Reply:
fixed